### PR TITLE
Misc/refactor SDK tests

### PR DESCRIPTION
**Pull request by samijaber**
## Description

cleanup the logic around our SDK tests. we had tons of arrays modifying test behaviour, now its all consolidated in the same object with the content JSON

### CODE REVIEW

File: packages/sdks-tests/src/specs/index.ts [hunk 1]
```diff
@@ -74,110 +74,119 @@ function isBrowser(): boolean {
...
+  '/custom-components-nowrap': { content: CUSTOM_COMPONENTS_NOWRAP },
+  '/override-base-url': { content: HTTP_REQUESTS },
 } as const;
 
 export type Path = keyof typeof PAGES;
 
-const GEN1_ONLY_PATHNAMES: Path[] = ['/api-version-v1', '/personalization-container'];
-const GEN2_ONLY_PATHNAMES: Path[] = [];
-
 export const getAllPathnames = (target: 'gen1' | 'gen2'): string[] => {
```
Comment (sidmohanty11) [inline packages/sdks-tests/src/specs/index.ts:182]: target can be of type `Page['target']` to include `'all'` what do you think?

File: packages/sdks-tests/src/specs/index.ts [hunk 2]
```diff
@@ -74,110 +74,119 @@ function isBrowser(): boolean {
...
+   * Defaults to `all`.
+   */
+  target?: 'gen1' | 'gen2' | 'all';
+  /**
+   * To test visual editing in Gen 1 SDK, we cannot provide a hardcoded JSON.
+   * Instead, we have to let the SDK fetch the data from the API and mock the
+   * network request instead.
+   * NOTE: This is why I can only test visual editing for the old SDK when using a non-SSR app
+   */
+  isVisualEditingTest?: boolean;
```
Comment (sidmohanty11) [inline packages/sdks-tests/src/specs/index.ts:93]: @samijaber can we name this specifically targeted to gen1, something like `isGen1VisualEditingTest` or anything similar? As there are other visual editor tests for columns or symbols where we only have them for gen2

Comment (samijaber) [reply packages/sdks-tests/src/specs/index.ts:93]: mm sure. we can rename it to `isVisualEditingTest` once all the visual editing tests have been updated to run on gen1

File: packages/sdks-tests/src/specs/index.ts (hunk 1 above)
Comment (samijaber) [reply packages/sdks-tests/src/specs/index.ts:182]: `getAllPathnames` is meant to be called within a specific e2e test app, to retrieve the pathnames that it needs to pre-render. so it doesn't make sense to allows `'all'` as an option

Comment (sidmohanty11) [reply packages/sdks-tests/src/specs/index.ts:182]: ohh right, makes sense

File: packages/sdks-tests/src/specs/index.ts (hunk 2 above)
Comment (sidmohanty11) [reply packages/sdks-tests/src/specs/index.ts]: yes! thank you

**Review by sidmohanty11 — APPROVED** [review]

**Pull request merged**